File: AutoStock.Domain/Entities/Supplier.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Runtime.ConstrainedExecution;
using System.Text;
using System.Threading.Tasks;

namespace AutoStock.Domain.Entities
{
    public class Supplier
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [EmailAddress]
        [MaxLength(150)]
        public string? Email { get; set; }

        [Phone]
        [MaxLength(20)]
        public string? PhoneNumber { get; set; }

        [MaxLength(250)]
        public string? Address { get; set; }

        public ICollection<Car> Cars { get; set; }
            = new List<Car>();

        public ICollection<PurchaseOrder>
        PurchaseOrders
            { get; set; }
            = new List<PurchaseOrder>();
    }
}

File: AutoStock.Client/src/types/supplier.ts
export interface Supplier {
    id: number;
    name: string;
    email: string | null;
    phoneNumber: string | null;
    address: string | null;
  }
  
  export interface SupplierRequest {
    name: string;
    email?: string | null;
    phoneNumber?: string | null;
    address?: string | null;
  }